### PR TITLE
NDRS-625: get-block-transfers rpc endpoint

### PR DESCRIPTION
**Pull request by EdHastingsCasperLabs**
This PR adds the get-block-transfers rpc endpoint and client subcommand, allowing external consumers to easily get all transfers for a given block (if any).

https://casperlabs.atlassian.net/browse/NDRS-625

```
{
  "jsonrpc": "2.0",
  "result": {
    "api_version": "1.0.0",
    "block_hash": "d577f98843c2b94b5e7e241aa8005b3ea86188211d10f702db684998143e4f70",
    "transfers": [
      {
        "amount": "100000000",
        "deploy_hash": "ab87c5f2c0f6f331bf488703676fb0c68f897282dfbb8e085752f220a3dfc25e",
        "from": "account-hash-1ace33e66142d5a0679ba5507ef75b9c09888d1567e86100d1db535fa819a962",
        "gas": "0",
        "id": null,
        "source": "uref-21f7316e72d1baa7b706a9083077d643665ad3a56673c594db9762ceac4f3788-007",
        "target": "uref-c5eb9788156b53c9a599dfb5e591c6399580b491c72086a6bc028dd18fdfcb2d-004"
      }
    ]
  },
  "id": 7229488934468542904
}
```

### CODE REVIEW

File: node/src/components/rpc_server/rpcs/chain.rs [hunk 1]
```diff
@@ -119,6 +130,103 @@ impl RpcWithOptionalParamsExt for GetBlock {
...
+    pub block_identifier: BlockIdentifier,
+}
+
+impl DocExample for GetBlockTransfersParams {
+    fn doc_example() -> &'static Self {
+        &*GET_BLOCK_TRANSFERS_PARAMS
+    }
+}
+
+/// Result for "chain_get_block" RPC response.
```
Comment (Fraser999) [inline node/src/components/rpc_server/rpcs/chain.rs:147]:
```suggestion
/// Result for "chain_get_block_transfers" RPC response.
```

**Review by dwerner — APPROVED** [review]
LGTM, the only suggestion I have is adding an integration test.

**Review by Fraser999 — COMMENTED** [review]
It looks like the only missing piece is the addition of the new RPC to the list of RPCs documented in `OPEN_RPC_SCHEMA` around [here](https://github.com/CasperLabs/casper-node/blob/master/node/src/components/rpc_server/rpcs/docs.rs#L66-L78).

**Pull request merged**